include/hipSYCL/runtime/hardware.hpp

```diff
@@ -60,6 +60,7 @@ enum class device_uint_property {
   max_global_size0,
   max_global_size1,
   max_global_size2,
+  max_group_sizes_need_flip,
```

Comment:
Is this specific to `max_group_sizes`? It also flips the work item indices passed into the kernel, global ranges etc. So maybe just `flips_dimensions` or `needs_dimension_flip` or something like that?

Reply:
True, I've opted for `needs_dimension_flip`.
So I expect there's other places, e.g. kernel launches, that need tweaking, can you point me to those?

Reply:
Not really. The relevant code would be the kernel launchers (inside `glue`), but those are already backend-specific and know the right behavior for their backend.
The one exception is the sscp kernel launcher, but that one currently only supports backends that do the flip, and so it just flips. Even then I'm not sure it would be possible to have the SSCP compilation flow flip only in some cases, since the assignment of builtin queries to indices would need to change which is handled inside the SYCL headers, so outside the control of the SSCP kernel launcher.

Reply:
Okay, so do you still think it's worth having this property at all?
I guess it makes sense if you can think of a backend out there that doesn't do the flip and whose limits aren't the same for every dimension.

Otherwise, I'm happy if you are with this PR. :-)